app/interactors/outdated_problem_clearer.rb

```diff
@@ -29,6 +29,12 @@ def compact_database
     collections = Mongoid.default_client.collections
     collections.map(&:name).map do |collection|
       Mongoid.default_client.command compact: collection
+    rescue Mongo::Error::OperationFailure => e
+      if e.message =~ /CMD_NOT_ALLOWED: compact/
+        next
```

Comment:
i guess the `next` is not strictly required here because there are no lines following this inside the loop? but maybe it's safer like this for the future?

Reply:
I was inclined to drop a return instead but wondered if other failures can occur for different collections - I don't have an authoritative answer. I'll lean to the team's better judgment instead.

Reply:
ah, sure, that would be a third option. and it probably does make sense to `return` and stop trying to compact collections if compact is not allowed, but the way you have it is fine.